### PR TITLE
plat-stm32mp1: SCMI service for clock and reset controllers

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by jforissier** [comment]
- Commit "plat-stm32mp1: fondation for scmi service":
`s/fondation/foundation/`
`s/everal/several/`

Minor: in commit sibjects, abbreviations should be uppercase: SCMI, SIP, SMC.

**Comment by etienne-lms** [comment]
> Commit "plat-stm32mp1: fondation for scmi service":
s/fondation/foundation/
s/everal/several/

Strange that my checkpatch tests does not report these.
Thanks anyway.

> Minor: in commit sibjects, abbreviations should be uppercase: SCMI, SIP, SMC.

Ok.

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+	for (i = 0; i < ARRAY_SIZE(agent_resources); i++) {
+		const struct scmi_agent_resources *res = &agent_resources[i];
+
+		for (j = 0; j < res->clock_count; j++) {
+			struct stm32_scmi_clk *clk = &res->clock[j];
+
+			if (!clk->name ||
+			    strlen(clk->name) >= SCMI_CLOCK_NAME_SIZE) {
+				EMSG("SCMI clock name invalid");
+				panic();
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/scmi_server.c:463]: `panic("SCMI clock name invalid");`

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+				stm32_clock_enable(clk->clock_id);
+		}
+
+		for (j = 0; j < res->rd_count; j++) {
+			struct stm32_scmi_rd *rd = &res->rd[j];
+
+			if (!rd->name ||
+			    strlen(rd->name) >= SCMI_RD_NAME_SIZE) {
+				EMSG("SCMI reset domain name too long");
+				panic();
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/scmi_server.c:478]: `panic("SCMI reset domain name too long");`

File: core/arch/arm/plat-stm32mp1/scmi_server.c (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/scmi_server.c:463]: oh true! I forgot you recently made changes to get panic logs on non debug configs.
I will fix all of them. Thanks to point out.

**Comment by etienne-lms** [comment]
comments addressed.
tag applied.

**Comment by etienne-lms** [comment]
Note: I left "SiP" with a lowercase "i" as it is how it's referred to in the Arm SMCCC specification.

**Pull request merged**